### DISCUSSION TITLE
VCPKG with embedded development?

### DISCUSSION THREAD

**Discussion by plimy** [microsoft/vcpkg#40121] [Q&A]
I have an embedded system with multiple processors, different architectures, some are 32 bit, some are 64 bit. And 1 is its own special bundle of joy.

I am contemplating moving our package management system to vcpkg (its this or conan, but my experience with vcpkg for home hobby packages makes me inclined towards vcpkg)

Is there a good way to setup the project so that I can have a top-level project that would pull in firmware packages for individual cores as dependencies, and have it compile each one (and all of that dependencies packages) with a cross compiler/compiler before bundling them altogether?

To make it more clear..

Say I have a port file for processing core A, which has a list of dependencies. I want core A to compile all of its dependencies using cross compiler A with cross compiler A flags.

While B, which has some shared dependencies with core A will need to compile all of its dependencies using cross compiler B with cross compiler B flags.
etc.

And I would like... to have core A/core B pulled down as a dependency for my platform.

**Comment by AugP** [comment]
Fundamentally, you need to set up custom triplets to configure how to build each configuration: https://learn.microsoft.com/vcpkg/concepts/triplets

If you set up vcpkg ports for Core A and Core B, they would each have a vcpkg.json manifest file where you can configure their own dependencies. Then for your top-level project, you should have another vcpkg.json file declaring your dependencies on Core A and Core B. Both triplets and manifest files are configurable so you can set various conditions, see reference documentation below: 
https://learn.microsoft.com/vcpkg/users/triplets
https://learn.microsoft.com/vcpkg/reference/vcpkg-json
